### DISCUSSION TITLE
Power indicator LED GPIO4 - can't invert (Sonoff Mini R2)

### DISCUSSION THREAD

**Discussion by sourcesimian** [arendst/Tasmota#16327] [Q&A] [answered]
I have built a Sonoff Mini R2 (`Module 1`) into an enclosure to operate as a IoT electrical socket. I connected a LED (with 10k resistor) to the S1/S2 terminals as an external power state indicator. (I soldered an external micro switch in parallel with the on-board micro switch, as the external push button). The LED toggles when the power is toggled, but in inverse - illuminated when relay is OFF, off when the relay is ON. I have tried both `GPIO4 288` and `GPIO4 320` strangely they give the same result. I must be missing something? I would appreciate any advice.

**Comment by barbudor** [comment]
I'm not sure to fully understand what you've done but S1 and S2 are intended to be used for an external switch, not for a LED.
You cannot assume that the internal circuitry of the Sonoff allows anything else than switches to be connected and controlled from the ESP

**Reply by sourcesimian** [reply]
Agreed that S1/S2 is marketed by Sonoff as the "external switch input", however it looks like the S1/S2 terminals map to GND/GPIO4 of the ESP8266, see: https://tasmota.github.io/docs/devices/Sonoff-Mini/. Thus, the expectation is that S2 can be used a Tasmota peripheral: https://tasmota.github.io/docs/Peripherals/. The `GPIO4 288`/`GPIO4 320` settings do change the behaviour to drive a LED, but they both default to ON?

**Reply by barbudor** [reply]
Your are assuming quickly but it may be dependant on the electronic circuit between the S2 pole and the GPIO.
Should I design such a device, I would opto-isolate it

Nevertheless, I understand you can make the led toggle but changing between Led1 and Led1_i have no impact ?
Do you still have the LedLink defined ?

**Reply by barbudor** [reply]
Looking at the photo, S2 seems to go through resistor and probably to the Q2 transistor
Definitively not sure you can drive this pin as an output
You may even create some conflicts between the ESP and the reste of the circuit

![image](https://user-images.githubusercontent.com/2996042/186436631-cde2cc1a-3b06-49d5-9a1c-71b843f95c47.png)

**Reply by sfromis** [reply]
If you want to verify how it works, try configuring it as Relay / Relay_i, and toggle it on/off.

**Reply by sourcesimian** [reply]
Thank you @sfromis, setting `GPIO4 224` (_"GPIO4 = Relay1"_) makes my external LED follow the power state. So the wiring is good, now to just figure out the right configuration. It seems weird to me that `Led_i` does not behave as the inverse of `Led`.

@barbudor my config does not use `LedLink`, should I?

**Reply by barbudor** [reply]
It should as it is specified in the template : https://templates.blakadder.com/sonoff_MINIR2.html
When not using a dedicated LedLink, Led1 behave as a combination of Power indicator and Link indicator
However that do not explain why Led_i do not behave as the opposite of Led. May be a side effect of not having a LedLink...

Still I'm not confident that you can safely turn the direction of the pin. It would be better to try figuring out the exact electronic circuit between S2 and GPIO14.

**Reply by sourcesimian** [reply]
Thank you for your time and help @barbudor. I have found a way to produce the desired behaviour using Rules, see below.

**Comment by sourcesimian** [comment, marked as answer]
This feels clunky, but it does produce the desired behaviour of an external LED on S1/S2 following the power state of the internal relay:
```
GPIO4 225  # GPIO4 = Relay2
Rule1 ON Power1#state==1 DO Power2 1 ENDON
Rule2 ON Power1#state==0 DO Power2 0 ENDON
```
Note: the MQTT topic will then be `.../POWER1` not `.../POWER`
